scripts/start/prepare_and_run.R

```diff
@@ -959,6 +959,9 @@ run <- function(start_subsequent_runs = TRUE) {
         file.copy("fulldata.gdx", paste0(cfg$gms$cm_CES_configuration,".gdx"), overwrite = TRUE)
         file.copy("fulldata.gdx", sprintf("input_%02i.gdx", cal_itr),
                   overwrite = TRUE)
+        if (file.exists("non_optimal.gdx")) {
+          file.rename("non_optimal.gdx", sprintf("non_optimal_%02i.gdx", cal_itr))
+        }
```

Comment:
How about consolidating this a bit?
```
interesting_output <- c('full.lst', 
                        'full.log', 
                        'fulldata.gdx', 
                        'non_optimal.gdx', 
                        'abort.gdx')

file.copy(from = interesting_output,
          to = sub('^(.*)(\\.[^\\.]+)$', 
                   sprintf('\\1_%02i\\2', cal_itr), 
                   interesting_output),
          overwrite = TRUE)
```
`file.copy()` doesn't care about missing files, just returns `FALSE` for those.

The 
```
file.copy("fulldata.gdx", paste0(cfg$gms$cm_CES_configuration,".gdx"), overwrite = TRUE)
```
part I would prefer to have somewhere outside the loop.  We should only provide this file if the calibration was successful (i.e. did all the calibration iterations set).  See below.

Reply:
Hmm, can't comment on lines you didn't touch.
So after the closing brace in line 972, before
```
  } else {
    stop("unknown realisation of 29_CES_parameters")
  }
```
add
```
# if the last CES calibration iteration was successful, write out calibration
# results
if (file.info("fulldata.gdx")$mtime > fulldata_m_time) {
    file.copy("fulldata.gdx", paste0(cfg$gms$cm_CES_configuration, '.gdx'))
    
    file.copy(from = paste0(cfg$gms$cm_CES_configuration, '_ITERATION_',
                            cfg$gms$c_CES_calibration_iterations, '.inc'),
              to = paste0(cfg$gms$cm_CES_configuration, '.inc'))
}
```

Reply:
> The
> 
> ```
> file.copy("fulldata.gdx", paste0(cfg$gms$cm_CES_configuration,".gdx"), overwrite = TRUE)
> ```
> 
> part I would prefer to have somewhere outside the loop. We should only provide this file if the calibration was successful (i.e. did all the calibration iterations set)

We already store a gdx named with the calibration-scenario which is later used for loading the right gdx-file, if the calibration was successful. Why do we need another gdx file?

Reply:
I suggested to create this file only when the calibration finishes successfully, not after each calibration iteration that had one or more feasible Nash iterations.  We don't want that file available for use when the calibration crashed half way.

Reply:
Your suggestion with the list to copy is not what I intend. I think `abort.gdx` and `non_optimal.gdx` should be renamed, not copied. Concerning the rest, I'm not really into calibration stuff and so it would be great if you could implement and test these changes yourself. So I would prefer to merge my stuff and you can improve it. But: should `non_optimal.gdx` and `abort.gdx` be renamed also for the last calibration run, or should they stay (or be copied) in this case such that the scripts correctly identify that there were problems?